Edited Date Picker Document (#2319)
vaadin/docs: articles/components/date-picker/index.asciidoc

----
--

=== Custom Validation

> Date Picker supports custom validation, such as limiting the options to Monday through Friday. In the example below, try to select a date that's on a Sunday or a Saturday. See how it reacts.

Reviewer: ```suggestion
Date Picker supports custom validation, such as limiting the options to Monday through Friday. In the following example, select a date that's on a Sunday or a Saturday to see a custom validation message.
```

----
--

== Initial Position

Date Picker's initial position parameter defines which date is focused in the calendar overlay when the overlay is opened. The default, initial position is the selected or current date.

> Use this feature to minimize the need for unnecessary navigation or scrolling when the user's input is expected to be within a certain time. Click on the example below and notice how it has a pulsating, rounded box around a particular date in the future, at the end of a quarter.

Reviewer: ```suggestion
Use this feature to minimize the need for unnecessary navigation or scrolling when the user's input is expected to be within a certain time. In the following example, click the Date Picker to open the calendar overlay and notice that the highlighted/focused date is in the future (the last day of the year), instead of the current date.
```

See also how to <<#date-format,configure a custom date format>>.

== Usage Patterns

=== Date Range

> You can create a date range picker using the Date Pickers twice. Imagine the example below is for an airline ticket booking page. It's asking the user for the date they want to depart and when they want to return. Try it: click on the calendar icon for the departure and select a date. Then click on the icon for the return date. Notice how dates prior to the departure date you chose are now disabled.

Reviewer: ```suggestion
You can create a date range picker using the Date Picker twice. Imagine the following example is for an airline ticket booking page. It's asking the user for the date they want to depart and when they want to return. Try it: select a departure date, and then proceed to select a return date. Notice how dates prior to the departure date you chose are now disabled.
```

To disable the days before the start date in the end date picker, you need to handle the selection in the start date picker and change the range in the end date picker.

== Best Practices

=== Picking vs. Typing

The calendar overlay is useful when the user needs to choose a day that's close to the current date or when information such as day of the week, week number, valid dates, and so on can aid in choosing the best option.

For far off dates (i.e., years ago or years from now) and for known dates (i.e., holidays and birthdays), typing the date in the input field can be faster and easier. Because of this, it's important to verify that the user can enter dates according to their locale.

> Instead of a Date Picker, you can use individual input fields for day, month, and year, to improve usability on small touch devices. In the example below, click on the year box and start to type a year. As you enter each digit, notice the list of choices narrows in the pull-down list.

Reviewer: ```suggestion
Instead of a Date Picker, you can use individual input fields for day, month, and year, to improve usability on small touch devices. In the following example, focus the year field and start to type a year. As you enter each digit, notice the list of choices narrows in the list.
```
